apps/www/src/app/create/index.tsx

```diff
@@ -1,27 +1,40 @@
-import React, { useState, FormEvent, ChangeEvent } from 'react';
-import Authenticate from '../shared/authenticate';
-import useTitle from '../shared/use-title.hook';
-import { User } from 'firebase/app';
+import React, { ChangeEvent, FormEvent, useEffect, useState } from 'react';
+import { useNavigate } from 'react-router-dom';
+import { User } from 'firebase/auth';
 import { Button, TextField } from '@material-ui/core';
-import { buildQuestion, Question } from '../shared/question';
-import { createTemplate } from '../shared/trivias.service';
-import SubmitError from './submit-error';
-import { useHistory } from 'react-router-dom';
 import QuestionForm from './question-form';
-
-import './index.scss';
-import { TriviaTemplate, buildTriviaTemplate } from '../shared/trivia';
 import Nav from '../nav';
+import Authenticate from '../shared/authenticate';
+import { Question, TriviaTemplate } from '../shared/common';
+import { buildDefaultQuestion } from '../shared/question';
+import { buildTriviaTemplate } from '../shared/trivia';
+import { createTemplate } from '../shared/trivias.service';
+import useTitle from '../shared/use-title.hook';
+import './index.scss';
 
 const CreateTriviaContent = ({ user }: { user: User }) => {
-  const history = useHistory();
+  const navigate = useNavigate();
+
+  const [error, setError] = useState<Error | null>(null);
   const [name, setName] = useState<string>('');
-  const [questions, setQuestions] = useState<Question[]>([buildQuestion()]);
-  const [error, setError] = useState<Error>(null);
+  const [questions, setQuestions] = useState<Question[]>([buildDefaultQuestion()]);
+  const [isDisabled, setIsDisabled] = useState<boolean>(true);
 
-  const handleAdd = () => {
-    setQuestions((questions) => questions.concat([buildQuestion()]));
-  };
+  useEffect(() => {
+    setIsDisabled(
+      !name || !questions ||
+        questions.some(
+          ({ possibleAnswers, correctAnswerIndex, question }: Question) =>
+            !question ||
+            !possibleAnswers ||
+            (correctAnswerIndex && correctAnswerIndex < 0) ||
+            possibleAnswers.some((answer) => !answer)
+        )
+    );
+  }, [name, questions]);
+
```

Comment:
sounds like you want `useMemo` instead of a new state
```suggestion
const isDisabled = useMemo(()=>
      !name || !questions ||
        questions.some(
          ({ possibleAnswers, correctAnswerIndex, question }: Question) =>
            !question ||
            !possibleAnswers ||
            (correctAnswerIndex && correctAnswerIndex < 0) ||
            possibleAnswers.some((answer) => !answer)
        )
    ), [name, questions]);
```